### DISCUSSION TITLE
달력 만들기

### DISCUSSION THREAD

**Discussion by soeunl** [soeunl/502-3-Study#28] [Show and tell]
package exam02과제;

 import java.time.LocalDate;
 import java.util.Scanner;

  public class Ex03 {
     public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);
        while (true) {
            System.out.println();
            System.out.println("년 월 입력 : ");
            String input = sc.nextLine(); // 2024 5

            if (input.equals("exit")) {
                System.out.println("종료합니다.");
                break;
            }

            try {
                String[] data = input.split("\\s+");
                int year = Integer.parseInt(data[0]);
                int month = Integer.parseInt(data[1]);

                System.out.printf("------ ♡%d년 %d월♡ ------%n", year, month);
                System.out.println("  일  월  화  수  목  금  토");
                getCalendar(year,month);
                System.out.println();
            } catch (Exception e) { // 숫자 형식 오류가 발생한 경우
                System.out.println("입력형식이 잘못되었습니다. 입력 예) 2024 5");
            }
        }
    }

    public static Object getCalendar(int year, int month) {
        /**
         * 1. 매월의 1일이 어떤 요일에 시작하는지
         * 2. 매월의 종료요일이 무엇인지(28일인지 29일인지 30일인지 31일인지)
         */
        LocalDate today = LocalDate.now(); // 현재 날짜 정보 가져오기
        //System.out.println(today);

        LocalDate startDate = LocalDate.of(year, month, 1); // 시작 날짜 지정
         //        System.out.println(startDate);

        LocalDate endDate = startDate.plusMonths(1).minusDays(1); // 끝 날짜 찾기
        //        System.out.println(endDate);

        int day = endDate.getDayOfMonth(); // 끝 날짜의 일을 정수로 day 변수에 담기

        int yoil = startDate.getDayOfWeek().getValue(); // 시작 날짜의 요일 번호 가져오기
        //        System.out.println(yoil);

        System.out.print("    ".repeat(yoil)); // 숫자 앞에 공백 넣기(요일의 숫자만큼)

        for (int i = 1; i <= day; i++) {

            if (i < 10) {
                System.out.print("  " + i + " ");
            }

            if (i >= 10) {
                System.out.print(" " + i + " ");
            }

            yoil++; // 시작 요일에서부터 요일의 숫자 증가
            if (yoil % 7 == 0) {
                System.out.println();
           }
        }
        return null;
    }
}

**Comment by heyejiyang** [comment]
```
package exam02;

import java.time.LocalDate;
import java.util.*;

public class CalEx {

    public static void main(String[] args) {
        //입력받는 부분 구현
        Scanner scan = new Scanner(System.in);
        //입력받는 문구
        String input = "";
        while (true){
            System.out.print("년 월 입력: ");
            input = scan.nextLine(); //2024 5 입력시 5월 달력 출력

            if(input.equals("exit")){
                System.out.println("프로그램을 종료합니다.");
                break;
            }

            //입력받은 문자가 형식에 맞지 않을경우 예외처리
            try {
                String[] data = input.split("\\s+"); //년도와 월 쪼개기
                int year = Integer.parseInt(data[0]);//String을 숫자형식으로 바꿈
                int month = Integer.parseInt(data[1]);

                System.out.printf(" ----- %d년 %d월 -----%n",year,month); //달력 출력

                //요일 출력 칸
                System.out.println(" S  M  T  W  T  F  S");
                //메서드 받아와서 출력시키기 ...
                getCalendar(year, month);

            }catch (Exception e){ //숫자 형식 오류가 발생한 경우
                System.out.println("입력 형식이 잘못되었습니다. 입력예시) 2024 5");
            }
        }
    }

    // 해당 달에 시작하는 1일이 몇요일인지 알아야한다
    // 해당 달에 끝나는 날짜를 알아야한다 28일, 29일, 30일, 31일
    public static void getCalendar(int year, int month){

        //년 월을 알고있으면 시작일자를 1로 고정
        LocalDate startDate = LocalDate.of(year, month,1);

        int yoil = startDate.getDayOfWeek().getValue() % 7; //요일의 값 가져오기 1(월) ~ 7(일) -> 0 ~ 6
       // yoil = yoil == 7 ? 0 :yoil;

        //5월 1일에서 다음달로 넘어가고(6월1일) -1 하면 마지막 날짜나옴
        LocalDate endDate = startDate.plusMonths(1).minusDays(1);
       // System.out.println(endDate); //2024-05-31
        int lastDay = endDate.getDayOfMonth();

        for(int i=1; i <= yoil; i++){
            System.out.print("   ");
        }
        for(int i = 1, n = yoil + 1; i <= lastDay; i++, n++){
           System.out.print((i<10)? "  "+ i:" "+i);

            if(n%7==0) System.out.println(); //토요일 지나면 줄개행
        }
        System.out.println();
    }

}

```
